src/main/java/bio/terra/service/dataset/flight/upgrade/predictableFileIds/ConvertToPredictableFileIdsUpdateMissingMd5ChecksumsStep.java

```diff
@@ -0,0 +1,69 @@
+package bio.terra.service.dataset.flight.upgrade.predictableFileIds;
+
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.filedata.google.firestore.FireStoreDao;
+import bio.terra.service.filedata.google.firestore.FireStoreFile;
+import bio.terra.service.filedata.google.gcs.GcsPdao;
+import bio.terra.service.filedata.google.gcs.GcsProjectFactory;
+import bio.terra.service.job.DefaultUndoStep;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.Storage;
+import java.util.List;
+import java.util.UUID;
+import liquibase.util.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConvertToPredictableFileIdsUpdateMissingMd5ChecksumsStep extends DefaultUndoStep {
```

Comment:
To make sure I understand -- even if the migration fails, if we've updated missing md5s in Firestore we want to keep them rather than roll them back.  That makes sense to me.

Reply:
Yeah exactly.  It was either this or add a whole other endpoint and that seemed like overkill